### ISSUE TITLE
Incorrect type inference in param.List.class_

### ISSUE THREAD

**Issue by vassilyl** [microsoft/pyright#4587]
**Describe the bug**
Incorrect type inferred.

**To Reproduce**
Install `param`:
  > pip install param
Run `pyright` against the code:
```python
import param
reveal_type(param.List.class_)
```
The output is
```
Found 1 source file
pyright 1.1.292
/workspaces/msrc-appconfig/checktypes.py
  /workspaces/msrc-appconfig/checktypes.py:3:13 - information: Type of "param.List.class_" is "Type[object]"
0 errors, 0 warnings, 1 information
```
**Expected behavior**
The proper type is `Unknown | None`. 

**Additional context**
Attribute `List.class_` refers to a slot that is initialized from an parameter without a type annotation and with default value of `None`:

```python
class List(Parameter):
    """
    Parameter whose value is a list of objects, usually of a specified type.

    The bounds allow a minimum and/or maximum length of
    list to be enforced.  If the item_type is non-None, all
    items in the list are checked to be of that type.

    `class_` is accepted as an alias for `item_type`, but is
    deprecated due to conflict with how the `class_` slot is
    used in Selector classes.
    """

    __slots__ = ['bounds', 'item_type', 'class_']

    def __init__(self, default=[], class_=None, item_type=None,
                 instantiate=True, bounds=(0, None), **params):
        self.item_type = item_type or class_
        self.class_ = self.item_type
```

**Comment by erictraut** [comment]
The `param` package is an untyped library. By default, pyright evaluates imported symbols from an untyped library as `Unknown`. I presume that you have enabled the `useLibraryCodeForTypes` setting in your configuration. When this is enabled, pyright will do its best to infer the type from the available code. Inference can be unreliable, which is why this feature is disabled by default. For type checking, we recommend that you leave this feature disabled. Type checking requires proper type information. If the `param` library is important to you, I recommend reaching out to its maintainers and suggesting that they add inlined type annotation. In the meantime, you might want to create your own type stub for the parts of the library that you use.

In this particular case, I'm not able to repro the behavior you're seeing. Perhaps you're using an older version of the library? I just installed the latest version, and pyright is inferring the type of `param.List.class_` as `Unknown | None` which is what you said you would expect.
